File: javascript/grade-school/grade-school.js
var School = function() {
    this.enrollment = {}
};

School.prototype.roster = function() {
    return this.enrollment;
}

School.prototype.add = function(name, grade_num) {
    if (this.enrollment[grade_num]) {
        this.enrollment[grade_num].push(name)
        this.enrollment[grade_num].sort()
    } else {
        this.enrollment[grade_num] = [name]
    }
}

School.prototype.grade = function(grade_num) {
    return (this.enrollment[grade_num]) ? this.enrollment[grade_num].sort() : []
}

module.exports = School;

File: ruby/grade-school/grade_school.rb
class School
  def initialize
    @students = Hash.new { |my_hash, key| my_hash[key] = [] }
  end

  def students(grade)
    @students[grade].sort
  end

  def add(student, grade)
    @students[grade] << student
  end

  def students_by_grade
    @students.keys.sort.map do |grade_level|
      { grade: grade_level, students: students(grade_level) }
    end
  end
end

module BookKeeping
  VERSION = 3
end
